airbyte_cdk/utils/docker/build.py

```diff
@@ -0,0 +1,412 @@
+
+    Returns:
+        The full image name with tag.
+
+    Raises:
+        FileNotFoundError: If the Dockerfile is not found.
+        subprocess.CalledProcessError: If the build fails.
+    """
+    dockerfile_path = connector_dir / "Dockerfile"
+    if not dockerfile_path.exists():
+        raise FileNotFoundError(f"Dockerfile not found: {dockerfile_path}")
+
+    image_name = metadata.dockerRepository
+    full_image_name = f"{image_name}:{tag}"
+
+    logger.info(f"Building Docker image from Dockerfile: {full_image_name} for platform {platform}")
+    logger.warning("Building from Dockerfile is deprecated. Consider using a base image in metadata.yaml.")
+
+    build_cmd = [
+        "docker",
+        "buildx",
+        "build",
+        "--platform",
+        platform,
+        "-t",
+        full_image_name,
+        "--label",
+        f"io.airbyte.version={metadata.dockerImageTag}",
+        "--label",
+        f"io.airbyte.name={metadata.dockerRepository}",
+        str(connector_dir),
+    ]
+
+    try:
+        run_docker_command(build_cmd)
+        logger.info(f"Successfully built image: {full_image_name}")
+        return full_image_name
+    except subprocess.CalledProcessError as e:
+        logger.error(f"Failed to build image: {e}")
+        raise
+
+
+def build_from_base_image(
+    connector_dir: Path,
+    metadata: ConnectorMetadata,
+    tag: str,
+    platform: str,
+) -> str:
+    """Build a Docker image for the connector using a base image.
+
+    Args:
+        connector_dir: Path to the connector directory.
+        metadata: The connector metadata.
+        tag: The tag to apply to the built image.
+        platform: The platform to build for (e.g., linux/amd64).
+
+    Returns:
+        The full image name with tag.
+
+    Raises:
+        ValueError: If the base image is not specified in the metadata.
+        subprocess.CalledProcessError: If the build fails.
+    """
+    if not metadata.connectorBuildOptions or not metadata.connectorBuildOptions.baseImage:
+        raise ValueError("Base image not specified in metadata.connectorBuildOptions.baseImage")
+
+    base_image = metadata.connectorBuildOptions.baseImage
+    image_name = metadata.dockerRepository
+    full_image_name = f"{image_name}:{tag}"
+
+    logger.info(f"Building Docker image from base image {base_image}: {full_image_name} for platform {platform}")
+
+    build_dir = connector_dir / "build" / "docker"
+    build_dir.mkdir(parents=True, exist_ok=True)
+
+    dockerfile_temp_path = build_dir / "Dockerfile"
+    dockerignore_temp_path = build_dir / ".dockerignore"
+
+    try:
+        try:
+            execute_build_customization_hooks(connector_dir, "pre_connector_install")
+        except subprocess.CalledProcessError:
+            logger.warning("Pre-install hook failed, continuing with build")
+
+        dockerfile_content = f"""
+FROM {base_image}
+
+WORKDIR /airbyte/integration_code
+
+COPY . .
+
+RUN pip install .
```

Comment:
Can you compare this with the logic in airbyte-ci's build? Did airbyte-ci just pip install directly from the local directory, or did it build and then install from the build?